server/src/entities/app_history.entity.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToOne,
+  JoinColumn,
+  Index,
+  Check,
+  BaseEntity,
+  CreateDateColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { App } from './app.entity';
+import { AppVersion } from './app_version.entity';
+import { User } from './user.entity';
+
+@Entity({ name: 'app_history' })
+@Index('IDX_UNIQUE_SEQ_PER_APP_VERSION', ['appVersionId', 'sequenceNumber'], { unique: true })
+@Check(
+  'check_history_payload_type',
+  `
+  (history_type = 'snapshot' AND jsonb_typeof(change_payload) = 'object') OR
+  (history_type = 'delta' AND jsonb_typeof(change_payload) = 'array')
+`
+)
+export class AppHistory extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ name: 'app_id', type: 'uuid' })
+  appId: string;
```

Comment:
app id can get from 
app_version -> app_id

won't be a good idea to add multiple relations